### DISCUSSION TITLE
Automatic query re-fetch after invalidating cache

### DISCUSSION THREAD

**Discussion by harshit-jainn** [urql-graphql/urql#2543] [Q&A] [answered]
I am using multiple fetch queries for different parts of a schema.
I am using @urql/exchange-graphcache and added a dummy mutatuion(without schema) to invalidate the root element in cache.
The cache is getting invalidated correctly, but the queries are not re-fetching data automatically for invalidated fields.
Currently i need to manually execute them.

In apollo we have this feature - https://www.apollographql.com/docs/react/data/refetching/#refetching-queries-affected-by-cache-updates

Do we have something similar to this in urql ?

**Comment by JoviDeCroock** [comment, marked as answer]
Normally when you invalidate a field a refetch would occur, as you don't have a schema this data can't be nullable so it should just refetch, mind reproducing this in a minimal way?

As mentioned in our [docs](https://formidable.com/open-source/urql/docs/graphcache/cache-updates/#invalidating-entities)

> in cases like these it may be advisable to trigger a refetch instead and let the cache update itself by sending queries that have invalidated data associated to them to our API again. This process is called invalidation since it removes data from Graphcache's locally cached data.

Our intention is a refetch.
